Add Hire Us CTA

dvdherron opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&CHANGE_ME)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->



## Description
Adds ability to add a Hire Us call to action at the bottom of pages. 


## Related Issue(s)
closes #805 



## Steps to test/reproduce
- See the bottom of the page of the two most recent articles (/2025/02/12/fluid-type/, and /2025/01/29/anchor-position-validity/
- And or add `hire_cta: true` to the bottom of the front matter on any post



## Show me
1. 
![image](https://github.com/user-attachments/assets/1235d3cf-927a-46cd-9bc9-98851f45f14d)

2.
![image](https://github.com/user-attachments/assets/b88b984a-7a08-442a-8437-441ba97e73c8)



To do:
- [x] fix colors in light mode

mirisuzanne commented on content/_includes/site/hire-us.njk:
```diff
@@ -0,0 +1,17 @@
+    <h2 id="cta-title">{{ title | mdInline | safe }}</h2>
+    {{ contents | md | safe }}
+
+  {{ layout.section_link("Let's chat", '/contact/') }}
```
I'm not sure this link has good contrast against the image. We could add a background to the link, or... Not sure if @SondraE has ideas.

dvdherron replied:
I changed this to look like a button  and made a new pattern for that if we think we'll have more link-buttons like this.

mirisuzanne commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,31 @@
+  --marker-color: var(--text);
+
+  background-image: url('/assets/media/transition-bg.png');
+  background-repeat: round;
```
The result here is a lot of stretching on the image - which would probably be fine if we removed the text from it. I think it could also work to bg-position `center`, and bg-size `cover`.

mirisuzanne reviewed (commented):
This is looking pretty good, but I left a few notes.

SondraE commented:
@dvdherron @mirisuzanne 
Reading through your review, Mia, and looking at the CTA section in context, I'm wondering whether something like this would be better. 

Less words.
Add a sample work image. (Not sure if the one I include in my mockup works.)
More contrast between the sizes of the text. 
Add a background to the CTA button.
Don't include the code in the background image.

<img width="1280" alt="Screenshot 2025-02-17 at 11 48 40 AM" src="https://github.com/user-attachments/assets/478166ce-17de-4541-8d83-077668e2d44b" />

SondraE reviewed (changes requested):
@dvdherron @mirisuzanne 
Reading through your review, Mia, and looking at the CTA section in context, I'm wondering whether something like this would be better. 

Less words.
Add a sample work image. (Not sure if the one I include in my mockup works.)
More contrast between the sizes of the text. 
Add a background to the CTA button.
Don't include the code in the background image.

<img width="1280" alt="Screenshot 2025-02-17 at 11 48 40 AM" src="https://github.com/user-attachments/assets/478166ce-17de-4541-8d83-077668e2d44b" />

dvdherron commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,31 @@
+  }
+
+  & :last-child {
+    @include config.font-family('body');
```
No, it's not overriding anything. But I'll still be more specific here and see if I can come up with a pattern for the title styles.

mirisuzanne replied:
oh, I see - we're just using two fonts. That makes sense.

dvdherron commented:
> Reading through your review, Mia, and looking at the CTA section in context, I'm wondering whether something like this would be better.
> 
> Less words. Add a sample work image. (Not sure if the one I include in my mockup works.) More contrast between the sizes of the text. Add a background to the CTA button. Don't include the code in the background image.

I like this stripped down version too. Do you have a version of the image somewhere without the code in the background @SondraE ?

dvdherron commented:
@mirisuzanne @SondraE I've addressed most of the issues in my latest commits. 

Remaining todos:
- change background image
- re-style link
- and probably some more code cleanup

I'll continue with the rest in the AM

dvdherron commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,31 @@
+  padding: var(--double-gutter) var(--page-margin) var(--double-gutter);
+}
+
+.hire-us-cta #cta-title {
```
drafted a patter for multi-line, multi-style titles in https://github.com/oddbird/oddleventy/pull/808/commits/e2fddfdc2fb3cc79ff87cbb5a62164da509334a6. Besides the main title on the homepage, are we doing this anyplace else?

dvdherron commented on content/blog/2025/anchor-position-validity.md:
```diff
@@ -19,6 +19,7 @@ summary: |
   If that doesn't work, step through the checklist,
   and then dive in to get a better understanding
   of how Anchor Positioning works.
+hire_cta: true
```
This is now changed so that the Hire Us cta is shown in all posts and OSS.

dvdherron commented on src/media/transition-bg.png:
@SondraE when you get a chance could you swap out the background image with the version with no text?

mirisuzanne commented on src/scss/layout/_header.scss:
```diff
@@ -62,3 +62,18 @@
   }
 }
+
+[data-header~='stacked'] {
```
I like this approach. I think great to use it on the home page as well. I'm not sure if we have stacked headers in other places. @SondraE might know.

mirisuzanne reviewed (approved):
looks good!

SondraE reviewed (changes requested):
Awesome!

Two more things:
1. Can you lay out the smaller screens sizes more like this?
<img width="714" alt="Screenshot 2025-02-18 at 10 49 44 AM" src="https://github.com/user-attachments/assets/2b0d0abb-eece-4996-ba7f-55c35f73ab34" />

2. Can you make the "Hire Us to Build" smaller relative to the "Custom Web Apps" text? I'm hoping for more of a feel like the text at the top of the homepage where there is a clear difference in size between the "We Create" text and the "Custom Web Apps" text.

SondraE commented:
@dvdherron Here the background image
![oddsite-hero](https://github.com/user-attachments/assets/0ef11805-a2fb-4491-b143-e354c113ce62)

stacyk commented on content/_includes/site/hire-us.njk:
```diff
@@ -0,0 +1,27 @@
+{% import 'embed.macros.njk' as embed %}
+
+{%- set title = '<span class="title-action">Hire Us to Build</span> Custom Web Apps <span class="title-description">That stand the test of time</span>' -%}
```
Should we keep Title Case throughout this whole line? I see the first  two lines are Title Case but the last line is sentence case. Is that intentional for emphasis?

dvdherron replied:
Yeah, I noticed that change on the second draft of the mockup @SondraE . Was that intentional?

SondraE replied:
@stacyk @dvdherron That was intentional. The "Hire Us to Build" matches what we have at the top of the homepage and the sentence case of the last line matches what we have on blog posts. We can consider something different if it doesn't work here.

stacyk commented on src/scss/initial/_links.scss:
```diff
@@ -68,3 +68,21 @@ a {
+// are removed and colors match styles applied
+// buttons
+[data-btn~='link-button'] {
+  --link: var(--bg);
+  --underline-thickness: 0px;
+  --underline-thickness--default: 0px;
+  --underline-color--default: transparent;
+  --underline-color: transparent;
+  --link-focus: var(--bg);
```
Do we need the `--defaults` here?

```suggestion
  --link: var(--bg);
  --link-focus: var(--bg);
  --underline-color: transparent;
  --underline-thickness: 0;
```

stacyk commented on src/scss/initial/_links.scss:
```diff
@@ -68,3 +68,21 @@ a {
+
+  &:hover,
+  &:focus {
+    --underline-thickness: 0px;
+    --underline-color: transparent;
```
```suggestion
    --underline-color: transparent;
    --underline-thickness: 0;
```

dvdherron replied:
fixed in https://github.com/oddbird/oddleventy/pull/808/commits/15b0a3f91ab66ebdb1aaf83febbce025309cdbfe

stacyk commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,40 @@
 }
+
+.hire-us-cta {
+  --reverse-y: 'content' 'image';
+  --media-gap: var(--double-gutter);
```
```suggestion
  --media-gap: var(--double-gutter);
  --reverse-y: 'content' 'image';
```

dvdherron replied:
fixed in https://github.com/oddbird/oddleventy/pull/808/commits/15b0a3f91ab66ebdb1aaf83febbce025309cdbfe

stacyk commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,40 @@
+  margin-block: var(--spacer);
+
+  @include config.above('wide-page') {
+    --cta-padding-block-end: var(--double-gutter);
+    --media-size: revert;
+    --align: center;
```
```suggestion
    --align: center;
    --cta-padding-block-end: var(--double-gutter);
    --media-size: revert;
```

stacyk commented on src/scss/layout/_cta.scss:
```diff
@@ -8,3 +10,40 @@
+  background-size: cover;
+  color: var(--text);
+  font-size: revert;
+  padding: var(--double-gutter) var(--page-margin)
+    var(--cta-padding-block-end, 0);
+  margin-block: var(--spacer);
```
```suggestion
  margin-block: var(--spacer);
  padding: var(--double-gutter) var(--page-margin)
    var(--cta-padding-block-end, 0);
```

dvdherron replied:
fixed in https://github.com/oddbird/oddleventy/pull/808/commits/15b0a3f91ab66ebdb1aaf83febbce025309cdbfe

stacyk reviewed (commented):
Looking good. I left a few small clean up items.

dvdherron commented:
> Awesome!
> 
> Two more things:
> 
>     1. Can you lay out the smaller screens sizes more like this?
> 

@SondraE Sure. With the first image, are you saying to put the image and the text side-by-side earlier?

> 
>     2. Can you make the "Hire Us to Build" smaller relative to the "Custom Web Apps" text? I'm hoping for more of a feel like the text at the top of the homepage where there is a clear difference in size between the "We Create" text and the "Custom Web Apps" text.

👍🏽

dvdherron commented:
@SondraE (and possibly @mirisuzanne too) I addressed Sondra's and Stacy's latest comments and this is ready for another round of review.

jgerigmeyer commented on content/_includes/site/hire-us.njk:
```diff
@@ -0,0 +1,27 @@
+
+<aside data-layout="cta" aria-labelledby="hire-cta-title" class="hire-us-cta" data-ccs-colors="dark">
+  {% set cta_content %}
+    <h2 id="hire-cta-title" data-header='stacked hire-cta'>{{ title | mdInline | safe }}</h2>
```
`title` isn't markdown, it's just HTML in this case -- is there a reason to `{% set %}` it instead of having it inline here?

dvdherron replied:
removed this in https://github.com/oddbird/oddleventy/pull/808/commits/b0426139237a6137623da2ce46314a44fe0de654

jgerigmeyer commented on content/_includes/site/hire-us.njk:
```diff
@@ -0,0 +1,25 @@
+          sizes='media'
+        )
+      ),
+      content=cta_content
```
If we use `{% call embed.media_block(...) %}` here, we could put `cta_content` inline instead of in a separate variable. Not a big deal, just might be more readable.

dvdherron replied:
fixed in https://github.com/oddbird/oddleventy/pull/808/commits/b0426139237a6137623da2ce46314a44fe0de654

stacyk commented on src/media/transition-bg-dark.png:
I think we can export these as jpg files and save a lot of data

stacyk replied:
@dvdherron maybe try these?

https://www.dropbox.com/scl/fi/ndk1abvk309zkqrwamwhq/callout-bg-1.jpg?rlkey=0a83dwoxak597o4gg42d8q648&dl=0

https://www.dropbox.com/scl/fi/x54sibnpvil29g0yfwebb/callout-bg-2.jpg?rlkey=u702k29m2l56crweaq0yau2tq&dl=0

stacyk commented on src/scss/initial/_links.scss:
```diff
@@ -68,3 +68,19 @@ a {
+
+  &:hover,
+  &:focus {
+    --underline-thickness: 0px;
+    --underline-color: transparent;
```
```suggestion
    --underline-color: transparent;
    --underline-thickness: 0;
```

dvdherron replied:
fixed in https://github.com/oddbird/oddleventy/pull/808/commits/d53ed4f2f9fad28423758e943101a785ca9de7f8

stacyk commented on src/scss/initial/_links.scss:
```diff
@@ -68,3 +68,19 @@ a {
+  &:hover,
+  &:focus {
+    --underline-color: transparent;
+    --underline-thickness: 0px;
```
```suggestion
    --underline-thickness: 0;
```

dvdherron replied:
Whoops. Missed that. Fixed in https://github.com/oddbird/oddleventy/pull/808/commits/a91c1fde653f587cca898006a504a8185a2da560
